editor/scaffolds/grid/wellknown/customer-grid.tsx

```
@@ -138,7 +524,54 @@ export function CustomerGrid({
   );
 
-  columns.unshift(SelectColumn);
+  // Custom header cell supporting indeterminate when estimatedCount is provided.
+  const selectColumn =
+    estimatedCount != null
+      ? {
+          ...SelectColumn,
+          renderHeaderCell: (props: RenderHeaderCellProps<unknown>) => {
+            const { column } = props;
+            // eslint-disable-next-line react-hooks/rules-of-hooks
+            const [, onRowSelectionChange] = useRowSelection();
```

Biome (2.3.13)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)